rhp/v4/merkle.go

```diff
@@ -0,0 +1,190 @@
+package rhp
```

Comment:
could this file just re-exported the stuff in rhp/v2?

Reply:
Probably a lot of it